include/integratorxx/quadratures/gausslegendre.hpp

```diff
@@ -93,20 +93,32 @@ struct quadrature_traits<
         // Old value of root
         point_type z_old = -2.0;
         // Values of P_n(x) and dP_n/dx
-        point_type p_n, dp_n;
+        point_type p_n, dp_n, p_n_m1;
 
         // Solve the root to eps absolute precision
-        while(std::abs(z-z_old) > eps) {
+        bool converged = false;
+        const int maxit = 100;
+        for(int it = 0; it < maxit; ++it) {
           // Evaluate the Legendre polynomial at z and its derivative
-          auto pn_eval = eval_Pn(z,npts);
-          p_n = pn_eval.first;
-          dp_n = pn_eval.second;
+          std::tie(p_n, dp_n, p_n_m1) = eval_Pn(z,npts);
```

Comment:
Since this is unused, you can pass `std::ignore`

Reply:
Thanks, again learning new things!